### PR TITLE
Throw a warning when WooCommerce is not active

### LINKED ISSUE #354: Notification when WooCommerce deactive and WooCommerce SEO plugin active
### How can we reproduce this behavior?
Since 11.1 when WooCommerce is **not active** and Yoast WooCommerce SEO **is active**, then WooCommerce SEO is not shown in SEO menu, comes from #351. 

It would be more user-friendly to give notification about it, as WooCommerce SEO can be activated but not shown anywhere and you cannot access it.

### Technical info
* Yoast SEO version: 11.1 RC1

### PR DESCRIPTION
**Pull request by jdevalk**
## Summary

This PR can be summarized in the following changelog entry:

* Throw a warning when WooCommerce is not active.
* Fix the wording for when Yoast SEO is not active.

## Relevant technical choices:

* Moved the `Yoast_WooCommerce_SEO` class from the main plugin file to its own file.
* Moved the checks and warnings inside the `Yoast_WooCommerce_SEO` class to their own `Yoast_WooCommerce_Dependencies` class.

## Test instructions

This PR can be tested by following these steps:

* Activate the plugin, make sure WooCommerce is _not_ active, see the warning.

Fixes #354 
Fixes https://github.com/Yoast/bugreports/issues/553

### CODE REVIEW

**Comment by jdevalk** [comment]
Will wait on all the other pulls currently open to be merged before doing this, as it would cause tons of merge conflicts otherwise.

File: Gruntfile.js [hunk 1]
```diff
@@ -10,6 +10,9 @@ module.exports = function( grunt ) {
 	const pkg = grunt.file.readJSON( "package.json" );
 	const pluginVersion = pkg.yoast.pluginVersion;
 
+	// Used to switch between development and release builds.
+	const developmentBuild = ! [ "release", "release:js", "artifact", "deploy:trunk", "deploy:master" ].includes( process.argv[ 2 ] );
```
Comment (jcomack) [inline Gruntfile.js]: This `= !` notation seems off. Shouldn't it be the other way around?

File: classes/woocommerce-dependencies.php [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+ *
+ * @package WPSEO/WooCommerce
+ */
+
+/**
+ * Class Yoast_WooCommerce_Dependencies
+ */
+class Yoast_WooCommerce_Dependencies {
+	/**
+	 * Check whether we've met our dependencies. If not, hook in some errors.
```
Comment (jcomack) [inline classes/woocommerce-dependencies.php]: Missing `@return` documentation.

File: classes/woocommerce-dependencies.php [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+	 */
+	public function check( ) {
```
Comment (jcomack) [inline classes/woocommerce-dependencies.php]: Unnecessary space between brackets.

File: classes/woocommerce-dependencies.php [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+		return false;
+	}
+
+	/**
+	 * Checks the dependencies. Sets a notice when requirements aren't met.
+	 *
+	 * @param string $wp_version The current version of WordPress.
+	 *
+	 * @return bool True whether the dependencies are okay.
```
Comment (jcomack) [inline classes/woocommerce-dependencies.php]: `True when` instead of `True whether`.

File: classes/woocommerce-dependencies.php [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+		}
+
+		// When WooCommerce is not installed.
+		if ( ! $this->check_woocommerce_exists() ) {
+			add_action( 'all_admin_notices', [ $this, 'woocommerce_missing_error' ] );
+
+			return false;
+		}
+
+		// When Yoast SEO SEO is not installed.
```
Comment (jcomack) [inline classes/woocommerce-dependencies.php]: Double `SEO` in comment.

File: classes/woocommerce-dependencies.php [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Checks whether WooCommerce is available.
+	 *
+	 * @return bool
```
Comment (jcomack) [inline classes/woocommerce-dependencies.php]: Missing `@return` description.

File: classes/woocommerce-dependencies.php [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+	 */
+	protected function check_woocommerce_exists() {
+		return function_exists( 'WC' );
+	}
+
+	/**
+	 * Returns the WordPress SEO version when set.
+	 *
+	 * @return bool|string The version whether it is set.
```
Comment (jcomack) [inline classes/woocommerce-dependencies.php]: `whether` -> `when`.

File: classes/woocommerce-dependencies.php [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+		return WPSEO_VERSION;
+	}
+
+	/**
+	 * Throw an error if WooCommerce is not active.
+	 */
+	public function woocommerce_missing_error() {
+		echo '<div class="error"><p>';
+		printf(
+		/* translators: %1$s resolves to the plugin search for Yoast SEO, %2$s resolves to the closing tag, %3$s resolves to Yoast SEO, %4$s resolves to WooCommerce SEO */
```
Comment (jcomack) [inline classes/woocommerce-dependencies.php]: Not sure how severe this is, but `%3$s` resolves to `WooCommerce` and `%4$s` resolves to `Yoast WooCommerce SEO`, not `WooCommerce SEO`.

File: classes/woocommerce-dependencies.php [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+		echo '</p></div>';
+	}
+
+	/**
+	 * Throw an error if WordPress SEO is not installed.
+	 */
+	public function yoast_wpseo_woocommerce_missing_error() {
+		echo '<div class="error"><p>';
+		printf(
+		/* translators: %1$s resolves to the plugin search for Yoast SEO, %2$s resolves to the closing tag, %3$s resolves to Yoast SEO, %4$s resolves to WooCommerce SEO */
```
Comment (jcomack) [inline classes/woocommerce-dependencies.php]: Not sure how severe this is, but `%4$s` resolves to `Yoast WooCommerce SEO` and not `WooCommerce SEO`.

File: classes/woocommerce-dependencies.php [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+
+	/**
+	 * Throw an error if WordPress is out of date.
+	 */
+	public function yoast_wpseo_woocommerce_wordpress_upgrade_error() {
+		echo '<div class="error"><p>';
+		printf(
+		/* translators: %1$s resolves to WooCommerce SEO */
+			esc_html__( 'Please upgrade WordPress to the latest version to allow WordPress and the %1$s module to work properly.', 'yoast-woo-seo' ),
+			'WooCommerce SEO'
```
Comment (jcomack) [inline classes/woocommerce-dependencies.php]: This seems inconsistent with the previous errors, which resolve to `Yoast WooCommerce SEO`.

File: classes/woocommerce-dependencies.php [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+	/**
+	 * Throw an error if WordPress SEO is out of date.
+	 */
+	public function yoast_wpseo_woocommerce_upgrade_error() {
+		echo '<div class="error"><p>';
+		printf(
+		/* translators: %1$s resolves to Yoast SEO, %2$s resolves to WooCommerce SEO */
+			esc_html__( 'Please upgrade the %1$s plugin to the latest version to allow the %2$s module to work.', 'yoast-woo-seo' ),
+			'Yoast SEO',
+			'WooCommerce SEO'
```
Comment (jcomack) [inline classes/woocommerce-dependencies.php]: This seems inconsistent with the previous errors, which resolve to `Yoast WooCommerce SEO`.

File: Gruntfile.js (hunk 1 above)
Comment (jdevalk) [reply Gruntfile.js]: This is the way it's implemented in all other plugins too :)

File: classes/woocommerce-dependencies.php (hunk 11 above)
Comment (jdevalk) [reply classes/woocommerce-dependencies.php]: Fixed.

File: classes/woocommerce-dependencies.php (hunk 10 above)
Comment (jdevalk) [reply classes/woocommerce-dependencies.php]: Fixed.

File: classes/woocommerce-dependencies.php (hunk 9 above)
Comment (jdevalk) [reply classes/woocommerce-dependencies.php]: Fixed.

File: Gruntfile.js (hunk 1 above)
Comment (jcomack) [reply Gruntfile.js]: Ah, okay. Nevermind then :)

**Comment by jcomack** [comment]
Acceptance done 👍

**Pull request merged**